java-checks-test-sources/src/main/java/checks/DitFrameworkExclusion.java

```diff
@@ -0,0 +1,23 @@
+package checks;
+
+import javax.swing.*;
+import org.eclipse.equinox.log.LogPermission;
+import org.springframework.dao.DataAccessException;
+
+public class DitFrameworkExclusion {
+  class MyJFrame extends JFrame {
+  }
+
+  class MyLogPermission extends LogPermission {
+    public MyLogPermission(String name, String actions) {
+      super(name, actions);
+    }
+  }
+
+  class MyDataAccessException extends DataAccessException {
```

Comment:
To me we still miss a test case where there is actually N levels BEFORE reaching an ignored class, and so we would raise an issue.